src/streamlink/plugins/vimeo.py

```diff
@@ -50,12 +50,8 @@ class Vimeo(Plugin):
         validate.any(None, validate.Schema(validate.get(1), _config_schema)),
     )
 
-    arguments = PluginArguments(
-        PluginArgument(
-            "mux-subtitles",
-            action="store_true",
-            help="Automatically mux available subtitles in to the output stream.",
-        )
+    global_arguments = PluginArguments(
```

Comment:
Instead of adding `global_arguments` what about adding a flag to `PluginArgument` to mark it as global. That way there is no extra `Plugin` parameter/attribute. The processing of the arguments would remain almost identical but I think the Plugin interface would be nicer.

Reply:
My initial idea was to not touch the Argument class, hence the `global_arguments` attribute, but I agree that this would be a better solution. Even better by subclassing the Argument class, so no parameter has to be set and the distinction becomes more clear. I will take a look at this later.